### PR TITLE
fix: [CDS-74650]: TagValue new filter scheme handling and fixed page breaks when using :value as tags

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
(#17325)

* fix: [CDS-74650]: TagValue new filter scheme handling and fixed page breaks when using :value as tags

* fix: [CDS-74650]: Test case to validate if tagValue retain its true value

* fix: [CDS-74650]: function rename and code cleanup

This also has qa-hf of - https://github.com/harness/harness-core-ui/pull/17370/commits/1445d092d8197e370c06d472d5ce4da6844877e0

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-74650]: https://harness.atlassian.net/browse/CDS-74650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDS-74650]: https://harness.atlassian.net/browse/CDS-74650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDS-74650]: https://harness.atlassian.net/browse/CDS-74650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vikrant-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/LogsContent/components/DownloadLogs/useDownloadLogs.tsx [hunk 1]
```diff
@@ -53,7 +53,8 @@ const checkStatusAndDownload = async (
         headers: {
           'X-Harness-Token': logsToken
         }
-      }
+      },
+      body: undefined
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/LogsContent/components/DownloadLogs/useDownloadLogs.tsx:57]: Why is this required? If body is not needed, don't pass it anything

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilter/ExecutionListFilter.tsx [hunk 2]
```diff
@@ -280,6 +283,14 @@ export function ExecutionListFilter(): React.ReactElement {
     }
   }
 
+  const pipelineTags = React.useMemo(
+    () =>
+      _pipelineTags?.reduce(
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilter/ExecutionListFilter.tsx:288]: Can we have another name instead of starting with `_`

Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListFilter/ExecutionListFilter.tsx:288]: sure, I will change it with develop PR (first)

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/70-pipeline/components/LogsContent/components/DownloadLogs/useDownloadLogs.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/components/LogsContent/components/DownloadLogs/useDownloadLogs.tsx:57]: Currently its required, but will check API spec as discussed
